models/DataObject/ClassDefinition/Data.php

```diff
@@ -1232,12 +1232,18 @@ public function adoptMainDefinition(DataObject\ClassDefinition\Data $mainDefinit
         }
     }
 
-    public function appendData(?array $existingData, array $additionalData): ?array
+    /**
+     * @throws \LogicException
+     */
+    public function appendData(?array $existingData, array $additionalData): array
```

Comment:
isn't the return type change a BC-break?

Reply:
It always returns array. We can keep `?array` for BC reason and add a TODO to change it in Pimcore 12.

Reply:
In case of custom classes extending this method (still having `?array`), it would crash due the declaration incompatibility

Reply:
I changed it back and add a TODO: https://github.com/pimcore/pimcore/pull/16458/commits/6281f48ae71ebbd27ef9b2bab0a21cad09e0bdee